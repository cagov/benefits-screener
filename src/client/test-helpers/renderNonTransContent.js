import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
