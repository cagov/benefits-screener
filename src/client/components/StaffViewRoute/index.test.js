import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import React from "react";
import Component from "./index";
import AUTH_STRINGS from "../../../data/authStrings";

const TestComponent = () => {
  return <div>TestComponent</div>;
};

describe("<StaffViewRoute />", () => {
  it("basic route", async () => {
    sessionStorage.removeItem(AUTH_STRINGS.authToken);
    const wrapper = renderNonTransContent(Component, "StaffViewRoute", {
      path: "/path",
      pageComponent: TestComponent,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("route in production (should be PageNotFound)", async () => {
    sessionStorage.removeItem(AUTH_STRINGS.authToken);
    const wrapper = renderNonTransContent(Component, "StaffViewRoute", {
      path: "/path",
      pageComponent: TestComponent,
      isProduction: true,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("route needing auth, but no data", async () => {
    sessionStorage.removeItem(AUTH_STRINGS.authToken);
    const wrapper = renderNonTransContent(Component, "StaffViewRoute", {
      path: "/path",
      pageComponent: TestComponent,
      pageProps: {
        userData: {},
        setUserData: () => {},
      },
      requiresAuthentication: true,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("route needing auth with user data", async () => {
    sessionStorage.removeItem(AUTH_STRINGS.authToken);
    const wrapper = renderNonTransContent(Component, "StaffViewRoute", {
      path: "/path",
      pageComponent: TestComponent,
      pageProps: {
        userData: { weeksToCertify: [0] },
        setUserData: () => {},
      },
      requiresAuthentication: true,
      computedMatch: {},
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("route needing auth with authToken", async () => {
    sessionStorage.setItem(AUTH_STRINGS.authToken, "TEST_TOKEN");

    // fetch isn't part of nodejs. If we pull in node-fetch, we
    // need to update the test codes to mock global.fetch.
    expect(global.fetch).toBeUndefined();
    const mockFetch = jest.fn(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve({
            status: AUTH_STRINGS.statusCode.ok,
            weeksToCertify: [0, 1],
          }),
      })
    );
    global.fetch = mockFetch;

    const wrapper = renderNonTransContent(Component, "StaffViewRoute", {
      path: "/path",
      pageComponent: TestComponent,
      pageProps: {
        userData: {},
        setUserData: () => {},
      },
      requiresAuthentication: true,
    });

    // Test cleanup.
    sessionStorage.removeItem(AUTH_STRINGS.authToken);
    global.fetch = undefined;

    expect(mockFetch.mock.calls.length).toBe(1);
    expect(mockFetch.mock.calls[0]).toEqual([
      AUTH_STRINGS.apiPath.data,
      {
        body: JSON.stringify({ authToken: "TEST_TOKEN" }),
        headers: {
          "Content-Type": "application/json",
        },
        method: "POST",
      },
    ]);

    expect(wrapper).toMatchSnapshot();
  });
});
