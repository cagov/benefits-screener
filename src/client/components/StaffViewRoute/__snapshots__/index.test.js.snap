// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffViewRoute /> basic route 1`] = `
<Route
  exact={true}
  path="/path"
>
  <Redirect
    push={true}
    to="/retroactive-certification"
  />
</Route>
`;

exports[`<StaffViewRoute /> route needing user data 1`] = `
<Route
  computedMatch={Object {}}
  exact={true}
  path="/retroactive-certification/staff-view/claimant-status"
>
  <Redirect
    push={true}
    to="/retroactive-certification"
  />
</Route>
`;

exports[`<StaffViewRoute /> route needing user data, without user data 1`] = `
<Route
  computedMatch={Object {}}
  exact={true}
  path="/retroactive-certification/staff-view/claimant-status"
>
  <Redirect
    push={true}
    to="/retroactive-certification"
  />
</Route>
`;
